File: src/main/java/vergisst/minecraftmod/weaponthrow/mixins/PlayerEntityMixin.java
package vergisst.minecraftmod.weaponthrow.mixins;

import net.minecraft.entity.player.PlayerEntity;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.Unique;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
import vergisst.minecraftmod.weaponthrow.Interface.IPlayerEntityMixin;
import vergisst.minecraftmod.weaponthrow.capabilities.PlayerThrowData;
import vergisst.minecraftmod.weaponthrow.events.OnStartPlayerTick;

@Mixin(PlayerEntity.class)
public class PlayerEntityMixin implements IPlayerEntityMixin {
    @Unique
    private PlayerThrowData throwPower = new PlayerThrowData((PlayerEntity)(Object)this);

    @Override
    public void weaponThrow$setThrowPower(PlayerThrowData value) {
        throwPower = value;
    }

    @Override
    public PlayerThrowData weaponThrow$getThrowPower() {
        return throwPower;
    }

    @Inject(method = "tick", at = @At("HEAD"))
    private void init(CallbackInfo info) {
        OnStartPlayerTick.EVENT.invoker().interact((PlayerEntity)(Object)this);
    }
}

File: src/main/kotlin/vergisst/minecraftmod/weaponthrowlite/mixins/PlayerEntityMixin.kt
package vergisst.minecraftmod.weaponthrowlite.mixins

import net.minecraft.entity.player.PlayerEntity
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.injection.At
import org.spongepowered.asm.mixin.injection.Inject
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo
import vergisst.minecraftmod.weaponthrowlite.api.IPlayerEntityMixin

import vergisst.minecraftmod.weaponthrowlite.capabilities.PlayerThrowData
import vergisst.minecraftmod.weaponthrowlite.events.OnStartPlayerTick


@Mixin(PlayerEntity::class)
abstract class PlayerEntityMixin: IPlayerEntityMixin {
    private var throwPower: PlayerThrowData = PlayerThrowData(this as Object as PlayerEntity)

    override fun setThrowPower(value: PlayerThrowData) {
        throwPower = value
    }

    override fun getThrowPower(): PlayerThrowData {
        return throwPower
    }

    @Inject(at = [At("HEAD")], method = ["tick"])
    fun init(info: CallbackInfo) {
//		throwPower =
        OnStartPlayerTick.EVENT.invoker().interact(this as Object as PlayerEntity)

//		println("Interface Path is ${IPlayerEntityMixin::class.java.`package`.name}")
    }
}